app/Http/Controllers/CartController.php

```diff
@@ -59,40 +65,155 @@ public function index(Request $request)
 	public function quickEntry(Request $request)
     {
 		$user = Auth::user();
+		$cart=Cart::with('User','CartItem.Product.Media')->where('user_id', $user->id)->get();
+		return view('quick-entry', array(
+				'cart' => $cart,
+			));
+			
+	}
+  
+	/**
+     * update cart item details
+     */
+    public function updateCartItem(Request $request)
+    {
+		$user = Auth::user();
+		if($request->has('cart_item_id')){
+			$cartitems=CartItem::where('id', $request->cart_item_id)->first();
+			if($cartitems){
+				$cart=Cart::where('id', $cartitems->cart_id)->first();
+					if($cart){
+						
+						if($request->option=="increment"){
+							$cart_quantity=$cart->total_items+1;
+							$cartitems->update(['quantity' => $cartitems->quantity+1]); 
+						}elseif($request->option=="decrement"){
+							$cart_quantity=$cart->total_items-1;
+							 
+							if($cartitems->quantity-1==0){
+								CartAttribute::where('cart_item_id', $cartitems->id)->delete();
+								$cartitems->delete(); 
+							}else{
+								$cartitems->update(['quantity' => $cartitems->quantity-1]);
+							}
+						}elseif($request->option=="delete"){
+							$cart_quantity=$cart->total_items-$cartitems->quantity;
+							CartAttribute::where('cart_item_id', $cartitems->id)->delete();
+							$cartitems->delete(); 
+						}	
+					    $cart->update(['total_items' => $cart_quantity]);						
+					}	
+				}
+		}
+		$cart=Cart::with('User','CartItem.Product.Media')->where('user_id', $user->id)->get();
 		
+
+		echo view('components.cart.product-list ', ['page' => 'cart','cart' => $cart]);
+	} 
+	
+	
+	/**
+     * update cart item marked_for details
+     */
+    public function updateCartItemMarked(Request $request)
+    {
+		if($request->has('cart_item_id')){
+			$cartitems=CartItem::where('id', $request->cart_item_id)->first();
+			if($cartitems){
+				$cartitems->update(['marked_for' => $request->marked_for]);
+			}
+		}
+		echo 1;
```

Comment:
@kams2391 why we are using echo here?

Reply:
Using this function into ajax.  That is why using echo. For now 1 is static value.

Reply:
Use json response here